test/integration/oauth2_test.rb

```diff
@@ -116,6 +161,11 @@ def validate_redirect(client, state)
   end
 
   def request_token(client, code, verifier = nil)
+    tokens = request_tokens(client, code, verifier)
+    tokens[:access_token]
+  end
+
+  def request_tokens(client, code, verifier = nil)
```

Comment:
I think we should leave this called `request_token` and get rid of the other routine, then have this just return `token` and have the callers pick out the bits they want.